### PR TITLE
Report memory and process information for devices.

### PR DESCRIPTION
**Pull request by cyberstip**

### CODE REVIEW

File: adb/contrib/high.py [hunk 1]
```diff
@@ -649,8 +649,46 @@ def GetBattery(self):
...
+    props = {}
+    out = self.PullContent('/proc/meminfo')
+    if not out:
+      return props
+
+    def parse_memline(p, key, res):
+      if line.startswith('%s:' % key):
+        parts = line.split()
+        if len(parts) == 3 and parts[2] == 'kB':
+          p[res] = int(parts[1])
```
Comment (maruel) [inline adb/contrib/high.py]: just use props directly here, it's a closure.

**Comment by cyberstip** [comment]
done

File: adb/contrib/high.py [hunk 2]
```diff
@@ -649,8 +649,46 @@ def GetBattery(self):
...
+
+  def GetProcessCount(self):
+    """Returns the number of total processes on the device."""
+    out, exit_code = self.Shell('ps')
+    if exit_code != 0:
+      return None
+    if not out.startswith('USER'):
+      return None
+
+    return len(out.splitlines()) - 1
```
Comment (maruel) [inline adb/contrib/high.py]: out.count('\n') -1 would be better hygiene. (Sorry I just thought about it)

**Comment by cyberstip** [comment]
ptal!

**Pull request merged**